src/wp-includes/user.php

```diff
@@ -682,14 +682,25 @@ function count_many_users_posts( $users, $post_type = 'post', $public_only = fal
 		return $pre;
 	}
 
-	$userlist = implode( ',', array_map( 'absint', $users ) );
-	$where    = get_posts_by_author_sql( $post_type, true, null, $public_only );
+	// Cleanup the users array. Remove duplicates and sort for consistent ordering.
+	$users = array_unique( array_filter( array_map( 'absint', (array) $users ) ) );
+	sort( $users );
 
-	$result = $wpdb->get_results( "SELECT post_author, COUNT(*) FROM $wpdb->posts $where AND post_author IN ($userlist) GROUP BY post_author", ARRAY_N );
+	$userlist    = implode( '_', $users );
+	$cache_key   = "count_many_users_posts_{$post_type}_{$public_only}_{$userlist}_" . get_current_user_id();
```

Comment:
Having a variable length cache key is problematic given Memcached has a [limit of 250 characters](https://github.com/memcached/memcached/blob/62b3447380d3fd547e432d1128d94cd12f2e6852/memcached.h#L67-L68) for a cache key. Granted, this normally won't be a problem, but if you're trying to get the posts for _many_ users, it can cause the cache key to go over this limit. Therefore, I suggest hashing the cache key. For example:

```suggestion
	$cache_key   = 'count_many_users_posts:' . md5( join( '_', array( $post_type, $public_only, get_current_user_id(), ...$users ) ) );
```

Reply:
Thank you, I've used @peterwilsoncc's suggestion using the `wp_cache_set_salted`